packages/block-editor/src/store/selectors.js

```diff
@@ -2814,11 +2814,7 @@ export function __unstableHasActiveBlockOverlayActive( state, clientId ) {
 	const editorMode = __unstableGetEditorMode( state );
 
 	// In zoom-out mode, the block overlay is always active for top level blocks.
-	if (
-		editorMode === 'zoom-out' &&
-		clientId &&
-		! getBlockRootClientId( state, clientId )
-	) {
+	if ( editorMode === 'zoom-out' && clientId ) {
```

Comment:
This should check that the block is either block level or child of sectionRootClientId if defined.

Reply:
Fixed in #59249.